Utils/Piece.py

```diff
@@ -1,10 +1,10 @@
 class Piece:
     def __init__(self, position):
         self.position = position
+        self.y, self.x = position
         self.has_moved = False
```

Comment:
How about creating some state memory for the piece? It should have a knowledge about `has_moved`, `last_move` etc.